src/Og.php

```diff
@@ -256,6 +256,30 @@ public static function getUserMemberships(AccountInterface $user, array $states
   }
 
   /**
+   * Returns the group membership for a given user and group.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   The user to get the membership for.
+   * @param \Drupal\Core\Entity\EntityInterface $group
+   *   The group to get the membership for.
+   * @param array $states
+   *   (optional) Array with the state to return. Defaults to active.
+   * @param string $field_name
+   *   (optional) The field name associated with the group.
+   *
+   * @return \Drupal\og\Entity\OgMembership|NULL
+   *   The OgMembership entity, or NULL if the user is not a member of the
+   *   group.
+   */
+  public static function getUserMembership(AccountInterface $user, EntityInterface $group, array $states = [OgMembershipInterface::STATE_ACTIVE], $field_name = NULL) {
+    foreach (static::getUserMemberships($user, $states, $field_name) as $membership) {
```

Comment:
@amitaibu I think we should leave this method as-is for now, as @pfrenssen wrote it. I think we can optimise it later. This keeps it simple, and keeps it statically cached too. If we load single stuff here we might want to look at how some stuff could be shared between getting all memberships and a single one. Maybe a lot of requests would need to get all memberships anyway?